### PR TITLE
mgmt, AKS, support AAD integration

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
### Features Added

- Supported disabling Kubernetes Role-Based Access Control for `KubernetesCluster` during create.
- Supported enabling Azure AD integration for `KubernetesCluster`.
- Supported disabling local accounts for `KubernetesCluster`.

### Other Changes

- Changed behavior that Linux profile is not required for `KubernetesCluster` during create.
 
Doc:
https://docs.microsoft.com/en-us/azure/aks/managed-aad
https://docs.microsoft.com/en-us/azure/aks/manage-azure-rbac

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by saadatbhatti** [comment]
No proper description what this PR is suppose to add

File: sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/BuiltInRole.java [hunk 1]
```diff
@@ -211,6 +211,29 @@
     public static final BuiltInRole KEY_VAULT_CRYPTO_SERVICE_ENCRYPTION_USER =
         BuiltInRole.fromString("Key Vault Crypto Service Encryption User");
 
+    // AKS related roles
+    /** Lets you manage all resources in the cluster. */
+    public static final BuiltInRole AZURE_KUBERNETES_SERVICE_RBAC_CLUSTER_ADMIN =
```
Comment (haolingdong-msft) [inline sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/BuiltInRole.java:216]: Where are those built-in roles used?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/BuiltInRole.java:216]: They are for user. Basically this class is for user.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/models/BuiltInRole.java:222]: Update javadoc for detail https://github.com/Azure/azure-sdk-for-java/pull/27729/commits/03016de9de6ec16edbba75f2b80d5d8f0888af2c

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java [hunk 2]
```diff
@@ -463,15 +472,50 @@
...
-//            WithCreate withRBACEnabled();
-//        }
+        /** The stage of the Kubernetes cluster definition allowing to specify Kubernetes Role-Based Access Control. */
+        interface WithRBAC {
+            /**
+             * Disables Kubernetes Role-Based Access Control.
+             *
+             * @return the next stage
+             */
+            WithCreate disableKubernetesRbac();
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:482]: What happens if KubernetesRbac is disabled? Does it mean user can only access the cluster using azure rbac?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:482]: I assume the admin/user KubeConfig should still work? They are the basics, unless be closed via `disableLocalAccouts`.

Azure RBAC requires Kubernetes RBAC. Cluster creation fails if this Kubernetes RBAC disabled.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java:482]: > AKS-managed Azure AD integration can't be disabled
> Changing a AKS-managed Azure AD integrated cluster to legacy AAD is not supported
> Clusters without Kubernetes RBAC enabled aren't supported for AKS-managed Azure AD integration

**Review by XiaofeiCao — APPROVED** [review]

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**